custom-recipes/nlp-analysis-ontology-tagging/recipe.json

```diff
@@ -0,0 +1,213 @@
+// This file is the descriptor for the Custom code recipe ontology-tagging
+{
+    // Meta data for display purposes
+    "meta": {
+       
+        "label": "Ontology tagging",
+        "description": "Assign given tags and to documents",
+        "icon": "icon-pushpin"
+    },
+
+    "kind": "PYTHON",
+   
+
+    "inputRoles": [
+        {
+            "name": "documents_to_tag",
+            "label": "Documents to tag",
+            "description": "the documents you wish to tag",
+            "arity": "UNARY",
+            "required": true,
+            "acceptsDataset": true
+        },
+        {
+            "name": "ontology_of_tags",
+            "label": "Ontology of tags",
+            "description": "dictionary with tags,keywords and categories",
+            "arity": "UNARY",
+            "required": true,
+            "acceptsDataset": true
+        }
+    
+    ],
+
+    "outputRoles": [
+        {
+            "name": "tagged_documents",
+            "label": "Tagged documents",
+            "description": "documents and their assigned tags",
+            "arity": "UNARY",
+            "required": true,
+            "acceptsDataset": true
+        }
+  
+    ],
+    
+    "selectableFromDataset": "documents_to_tag",
+    "selectableFromDataset":"ontology_of_tags",
+
+
+    "params": [
+        
+         {
+            "name": "separator_input",
+            "label": "Document dataset",
+            "type": "SEPARATOR"
+        },
+        
+        
+        {
+            "name": "text_column",
+            "label": "Text column",
+            "type": "COLUMN",
+            "description": "text column containing the documents",
+            "mandatory": true,
+            "columnRole": "documents_to_tag",
+            "allowedColumnTypes": [
+                "string"
+            ]
+        },
+        {                                           #TODO add a comment about the supported languages? 
+            "name": "language_column",
+            "label":'Language column',
+            "type": "COLUMN",
+            "mandatory":true,
+            "columnRole": "documents_to_tag",
+            "allowedColumnTypes": [
+                "string"
+            ]
+            
+        },
+        
+        {
+            "name": "tags_dataset",
+            "label": "Tags dataset",
+            "type": "SEPARATOR"
+        },
+        
+        {
+            "name": "tag_column",
+            "label": "Tag column",
+            "type":"COLUMN",
+            "mandatory":true,
+            "columnRole":"ontology_of_tags",
+            "allowedColumnTypes": [
+                "string"
+            ]
+        },
+        
+       
+         
+        {
+            "name":"keyword_column",
+            "label":"Keyword column",
+            "type":"COLUMN",
+            "mandatory":false,
```

Comment:
We've previously agreed that the user could simply give tags, which will imply that string matching will be done on tags instead of keywords. Have you changed your mind ? If you prefer we could say that the user have to give keywords no matter what : and then we will have keyword column mandatory.

Reply:
Good point, I didn't precisely remember that we had discussed this. I am fine with either. I had a second thought because I wonder if it would be intuitive for the end-user

Or to put it another way: if we keep this parameter optional, and the user forgets to fill the keywords column argument (but the column exists), wouldn't he/she be surprised by the results?

What do you think?

Reply:
I've made it mandatory, as it does not cost a lot for the user, and prevent the risks you highlited